### PR TITLE
ROTC Component

### PR DESCRIPTION
**Pull request by alexose**

### CODE REVIEW

File: src/js/edu-benefits/components/RotcHistoryFields.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            onValueChange={(update) => {this.props.onStateChange('seniorRotcIsParticipating', update);}}/>
+
+      </div>
+    </fieldset>
+    );
+  }
+}
+
+RotcHistoryFields.propTypes = {
+  onStateChange: React.PropTypes.func.isRequired,
```
Comment (jbalboni) [inline src/js/edu-benefits/components/RotcHistoryFields.jsx]: Add `initializeFields` here?

File: src/js/edu-benefits/utils/veteran.js [hunk 2]
```diff
@@ -47,6 +54,8 @@ export function createVeteran() {
     toursOfDuty: [],
     seniorRotcComissioned: makeField(''),
     seniorRotcComissionYear: makeField(''),
+    seniorRotcScholarships: [],
```
Comment (jbalboni) [inline src/js/edu-benefits/utils/veteran.js]: I think we want to match the json schema here if possible: https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/dist/edu-benefits-schema.json#L450

File: test/edu-benefits/components/RotcHistoryFields.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import SkinDeep from 'skin-deep';
+import { expect } from 'chai';
+
+import RotcHistoryFields from '../../../src/js/edu-benefits/components/RotcHistoryFields';
+import { createVeteran } from '../../../src/js/edu-benefits/utils/veteran';
+
+describe('<RotcHistoryFields>', () => {
```
Comment (jbalboni) [inline test/edu-benefits/components/RotcHistoryFields.unit.spec.jsx]: I think I've settled on only testing Fields components if they have conditional rendering logic in them. At least for the last few pages I've done. But I don't know what anyone else thinks about that.

File: src/js/edu-benefits/reducers/uiState/index.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ const ui = {
     '/military-history/rotc-history': {
       complete: false,
       verified: false,
-      fields: []
+      fields: ['seniorRotcIsParticipating']
```
Comment (jbalboni) [inline src/js/edu-benefits/reducers/uiState/index.js]: Does the scholarship list need to be in here?

File: src/js/edu-benefits/components/RotcScholarship.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            field={scholarship.amount}
+            placeholder="$"
+            onValueChange={(update) => {onValueChange('amount', update);}}/>
+        <ErrorableTextInput
+            errorMessage={validateIfDirty(scholarship.year, isValidYear) ? undefined : 'Please enter a valid year.'}
+            validation={validateIfDirty(scholarship.year, isValidYear)}
+            label="Year recieved"
+            name="year"
+            field={scholarship.year}
+            placeholder="YYYY"
```
Comment (jbalboni) [inline src/js/edu-benefits/components/RotcScholarship.jsx]: Are we using uppercase or lowercase for year placeholders?

File: src/js/edu-benefits/components/RotcHistoryFields.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            data={this.props.data}
+            initializeCurrentElement={() => this.props.initializeFields(scholarshipFields, 'seniorRotcScholarships')}
+            onRowsUpdate={(update) => {this.props.onStateChange('seniorRotcScholarships', update);}}
+            path="military-history/rotc-history"
+            rows={this.props.data.seniorRotcScholarships}
+            isValidSection={isValidSection}/>
+
+        <ErrorableRadioButtons
+            label="Are you currently participating in a senior ROTC scholarship program that pays for your tuition, fees, books and supplies under Section 2107 of Title 10, U.S. Code?"
+            options={['Yes', 'No']}
```
Comment (jbalboni) [inline src/js/edu-benefits/components/RotcHistoryFields.jsx]: There's a `yesNo` array in options-for-select.js that I've been using.

File: src/js/edu-benefits/components/RotcScholarship.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+import { isValidYear, validateIfDirty, isNotBlank } from '../utils/validations';
+
+export default class RotcScholarship extends React.Component {
+  render() {
+    const { view, onValueChange } = this.props;
+    const scholarship = this.props.data;
+    const formFields = (
+      <div>
+        <ErrorableTextInput
```
Comment (jbalboni) [inline src/js/edu-benefits/components/RotcScholarship.jsx]: Might be worth asking if any of these fields should be required. I think we normally have at least one in these sorts of lists.

File: src/js/edu-benefits/components/RotcHistoryFields.jsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      <p>ROTC placeholder</p>
+      <div className="input-section">
+
+        <GrowableTable
+            component={RotcScholarship}
+            createRow={createRotcScholarship}
+            data={this.props.data}
+            initializeCurrentElement={() => this.props.initializeFields(scholarshipFields, 'seniorRotcScholarships')}
+            onRowsUpdate={(update) => {this.props.onStateChange('seniorRotcScholarships', update);}}
+            path="military-history/rotc-history"
```
Comment (jbalboni) [inline src/js/edu-benefits/components/RotcHistoryFields.jsx]: Does this need to start with `/`?

**Comment by jbalboni** [comment]
@alexose This looks good to me, just made some nitpicky comments.

File: src/js/edu-benefits/utils/veteran.js (hunk 2 above)
Comment (alexose) [inline src/js/edu-benefits/utils/veteran.js]: TIL where this file lives :)

File: test/edu-benefits/components/RotcHistoryFields.unit.spec.jsx (hunk 3 above)
Comment (alexose) [inline test/edu-benefits/components/RotcHistoryFields.unit.spec.jsx]: That seems fine to me.  I thought I might test some of the `GrowableTable` functionality in here, but then I realized that it wasn't really part of this PR.

Removing for now.

File: src/js/edu-benefits/components/RotcScholarship.jsx (hunk 5 above)
Comment (alexose) [inline src/js/edu-benefits/components/RotcScholarship.jsx]: Let's go with lower case to match the DateInput element...

**Pull request merged**